ctera/src/main/java/ch/cyberduck/core/ctera/CteraAclPermissionFeature.java

```diff
@@ -0,0 +1,214 @@
+     * Files only.
+     */
+    public static final Acl.Role EXECUTEPERMISSION = new Acl.Role("executepermission");
+
+    /**
+     * Allows or denies deleting the file or folder. If you don't have Delete permission on a file or folder,
+     * you can still delete it if you have been granted Delete Subfolders and Files on the parent folder.
+     */
+    public static final Acl.Role DELETEPERMISSION = new Acl.Role("deletepermission");
+
+    /**
+     * Traverse Folder: Allows or denies moving through a restricted folder to reach files and folders
+     * beneath the restricted folder in the folder hierarchy. Traverse folder takes effect only when the group or user
+     * is not granted the "Bypass traverse checking user" right in the Group Policy snap-in.
+     * This permission does not automatically allow running program files.
+     * Directories only.
+     */
+    public static final Acl.Role TRAVERSEPERMISSION = new Acl.Role("traversepermission");
+
+    /**
+     * Create Files: Allows or denies creating files within the folder.
+     * Directories only.
+     */
+    public static final Acl.Role CREATEFILEPERMISSION = new Acl.Role("Createfilepermission");
+
+    /**
+     * Create Folders: Allows or denies creating subfolders within the folder.
+     * Directories only.
+     */
+    public static final Acl.Role CREATEDIRECTORIESPERMISSION = new Acl.Role("CreateDirectoriespermission");
+
+    static final List<Acl.Role> allCteraCustomACLRoles = Collections.unmodifiableList(Arrays.asList(
+            READPERMISSION, WRITEPERMISSION, EXECUTEPERMISSION, DELETEPERMISSION, TRAVERSEPERMISSION, CREATEFILEPERMISSION, CREATEDIRECTORIESPERMISSION
+    ));
+
+    final static List<QName> allCteraCustomACLQn = Collections.unmodifiableList(allCteraCustomACLRoles.stream().map(CteraAclPermissionFeature::toQn).collect(Collectors.toList()));
+
+    public static QName toQn(final Acl.Role role) {
+        return new QName(CTERA_NAMESPACE_URI, role.getName(), CTERA_NAMESPACE_PREFIX);
+    }
+
+    public static String toProp(final QName qn) {
+        return qn.getLocalPart();
+    }
+
+    public static Acl.Role toRole(final QName qn) {
+        return new Acl.Role(toProp(qn));
+    }
+
+    public static Acl customPropsToAcl(final Map<String, String> customProps) {
+        if(customProps.isEmpty()) {
+            return Acl.EMPTY;
+        }
+        final Acl acl = new Acl();
+        acl.addAll(new Acl.CanonicalUser());
+        for(QName qn : allCteraCustomACLQn) {
+            if(customProps.containsKey(toProp(qn))) {
+                final String val = customProps.get(toProp(qn));
+                if(Boolean.parseBoolean(val)) {
+                    acl.addAll(new Acl.CanonicalUser(), toRole(qn));
+                }
+            }
+        }
+        return acl;
+    }
+
+    public static Permission aclToPermission(final Acl acl) {
+        if(Acl.EMPTY.equals(acl)) {
+            return new Permission(700);
```

Comment:
@dkocher is this correct? See [comment](https://github.com/iterate-ch/cyberduck/pull/15468/files#r1533320170) on asymmetry.